docs(OAuth2):added guide for aws cognito (#2017)
spinnaker/spinnaker.github.io: setup/security/authentication/oauth/cognito/index.md

---
title:  "AWS Cognito"
sidebar:
  nav: setup
---

This page instructs you on how to obtain an OAuth 2.0 client ID and client secret for
use with your AWS Cognito User Pools.

## Setting up an AWS Cognito App Client

> 1. Navigate to [https://aws.amazon.com/](https://aws.amazon.com/) and log in with your AWS credentials.

Reviewer: Can we link directly to the Cognito page?

2. Search for Cognito in the search bar.
3. Select the user pools you want Spinnaker to use.

> 4. At the side bar under "General settings", select "App clients", add a client

Reviewer: ```suggestion
4. At the side bar under "General settings", select "App clients", add a client.
```

> 4. At the side bar under "General settings", select "App clients", add a client.

Reviewer: ```suggestion
4. At the side bar under **General settings**, select **App clients**, add a client.
```

Reviewer: Sorry, @joetancy, I should have done this the last time I looked at this. Can you change all UI elements to be in boldface instead of double-quotes? Thanks!

https://developers.google.com/style/ui-elements

>   - Make sure you select "Generate client secret"

Reviewer: ```suggestion
  - Make sure you select "Generate client secret."
```

> 5. After that go to "App integration", then to "App client settings"

Reviewer: ```suggestion
5. After that go to "App integration," then to "App client settings."
```

>   - Select "Cognito User Pool" as one of the "Enabled Identity Providers"

Reviewer: ```suggestion
  - Select "Cognito User Pool" as one of the "Enabled Identity Providers."
```

>   - Select "Cognito User Pool" as one of the "Enabled Identity Providers"

Reviewer: This one, and the ones following, should probably be numbered instead of sub-bullets. They're just further steps, right? (The two items _under_ "Check the following" are ok as bullets though, of course.

>   a) Select "Cognito User Pool" as one of the "Enabled Identity Providers."

Reviewer: @joetancy I don't know if you've set yourself up to preview your changes, but these substeps might not resolve the way you want them to. If you _are_ previewing, and this looks ok, then ignore this comment. 

But Markdown usually wants sub steps to start with `1. `, just like the first-level steps. And the indentation causes Markdown to give them letters instead of numbers.

Author: apologies, fixed in the next commit!

>   - Input your callback url

Reviewer: ```suggestion
  - Input your callback URL.
```

Reviewer: Periods for all these that are complete sentences.

>     - Check the following

Reviewer: ```suggestion
    - Check the following:
```

       - Authorization code grant, Implicit grant
       - email, openid

>     - Also make sure you already have a domain name for your hosted UI

Reviewer: ```suggestion
    - Also make sure you already have a domain name for your hosted UI.
```

> You should have these credentials ready before moving on to the next step

Reviewer: ```suggestion
Have these credentials ready before moving on to the next step
```

> Have these credentials ready before moving on to the next step

Reviewer: ```suggestion
Have these credentials ready before moving on to the next step:
```

- App client id
- App client secret
- Hosted UI domain name

## Configure Halyard

> You may configure Halyard either with the CLI or by manually editing the hal config.

Reviewer: ```suggestion
You can configure Halyard either with the [CLI](/reference/halyard/commands/) or by manually editing the hal config.
```

### Hal config

```yaml
security:
  authn:
      oauth2:
        enabled: true
        client:
          clientId: {CLIENT_ID}
          clientSecret: {CLIENT_SECRET}
          accessTokenUri: {YOUR_DOMAIN_NAME}/oauth2/token
          userAuthorizationUri: {YOUR_DOMAIN_NAME}/oauth2/authorize
          preEstablishedRedirectUri: {GATE_URL}/login
          useCurrentUri: false
        resource:
          userInfoUri: {YOUR_DOMAIN_NAME}/oauth2/userInfo
        userInfoMapping: {}
        provider: OTHER
```

### CLI

> Set up OAuth 2.0 with AWS Cognito:

Reviewer: ```suggestion
1. Set up OAuth 2.0 with AWS Cognito:
```

`hal config security authn oauth2 edit --provider OTHER --client-id (client ID from above)  --client-secret (client secret from above) --access-token-uri (your domain name)/oauth2/token --user-authorization-uri (your domain name)/oauth2/authorize --user-info-uri (your domain name)/oauth2/userInfo`

> Now enable OAuth 2.0 using hal:

Reviewer: ```suggestion
Enable OAuth 2.0:
```
